### PR TITLE
USD fixes

### LINKED ISSUE #1186: USD: Specific file is missing colors
**Describe the bug**
A specific (series of) files are missing colors: https://github.com/mwestphal/assets/blob/main/full_assets/Vehicles/USD_Mini_Car_Kit/assets/vehicles/vehicleVariants.usda

F3D: 

a

Blender:

a

### PR DESCRIPTION
**Pull request by Meakk**
Fix #1186 #1182 #1166  
Improves #1185 #1184

### CODE REVIEW

File: plugins/usd/module/vtkF3DUSDImporter.cxx [hunk 1]
```diff
@@ -290,6 +409,15 @@ class vtkF3DUSDImporter::vtkInternals
                   // interpolation is face-varying.
                   // It will be normalized by the vtkF3DFaceVaryingPointDispatcher later
                   newPolyData->GetPointData()->AddArray(texCoords);
+
+                  if (firstArray)
+                  {
+                    // sometimes we are enable to fetch the array name to use for texture mapping
+                    // so we fallback to the first UV set added
+                    // see https://github.com/f3d-app/f3d/issues/1184
+                    firstArray = false;
```
Comment (mwestphal) [inline plugins/usd/module/vtkF3DUSDImporter.cxx:418]: firstArray seems unused ?

File: plugins/usd/module/vtkF3DUSDImporter.cxx [hunk 2]
```diff
@@ -290,6 +409,15 @@ class vtkF3DUSDImporter::vtkInternals
...
+                    newPolyData->GetPointData()->SetTCoords(texCoords);
```
Comment (mwestphal) [inline plugins/usd/module/vtkF3DUSDImporter.cxx:419]: Ok I get it now. You set it here in case its not set later ?

**Review by mwestphal — APPROVED** [review]
Approved assuming my remark is correct

**Comment by mwestphal** [comment]
no idea what is happening with CI

File: plugins/usd/module/vtkF3DUSDImporter.cxx (hunk 2 above)
Comment (Meakk) [reply plugins/usd/module/vtkF3DUSDImporter.cxx:419]: Yes it can be empty in case we have a texture transform layer, but we do not support it yet.

Comment (Meakk) [reply plugins/usd/module/vtkF3DUSDImporter.cxx:419]: Yes it can be empty in case we have a texture transform layer, but we do not support it yet.

**Pull request closed without merging**